### PR TITLE
Prevent private key lose during Account key link

### PR DESCRIPTION
**Pull request by AnthonyLaw**
### Fix
- To store Remote and VRF key info before announcing the transaction, to prevent listener or web socket disconnect and lose the private key.
- Pre-store `selectedHarvestingNodes` when announcing link txs, and removed if supplement node pubkey is not exist

Resolved: #1476

### CODE REVIEW

**Comment by AnthonyLaw** [comment]
To continue to fix this issue, https://github.com/symbol/desktop-wallet/issues/1476#issuecomment-888589927
we need to fix the SDK first. https://github.com/symbol/symbol-sdk-typescript-javascript/issues/806

**Comment by cryptoBeliever** [comment]
@AnthonyLaw looks like harvesting for multisig stopped working:

https://share.getcloudapp.com/12uveLeR

File: src/store/Harvesting.ts [hunk 1]
```diff
@@ -328,40 +353,127 @@ export default {
             { commit },
             { accountAddress, selectedHarvestingNode }: { accountAddress: string; selectedHarvestingNode: NodeModel },
         ) {
-            const harvestingService = new HarvestingService();
+            console.log('selectedHarvestingNode :>> ', selectedHarvestingNode);
```
Comment (bassemmagdy) [inline src/store/Harvesting.ts:356]: could you remove logs?

File: src/store/Harvesting.ts [hunk 2]
```diff
@@ -328,40 +353,127 @@ export default {
...
+                newEncVrfPrivateKey,
+                newRemotePublicKey,
+                newVrfPublicKey,
+                encRemotePrivateKey,
+                encVrfPrivateKey,
+            } = currentSignerHarvestingModel;
+
+            // If remote key link not exist from network or local
+            // Set local encRemotePrivateKey to null
+            if (!linked || !newRemotePublicKey) {
```
Comment (bassemmagdy) [inline src/store/Harvesting.ts:443]: should be `&&` instead of `||`, no?

Comment (AnthonyLaw) [reply src/store/Harvesting.ts:443]: I believe use `||` is more suitable in this situation.
because if 1 of them is empty, just set to `null`

**Review by bassemmagdy — APPROVED** [review]

**Pull request merged**